sgkit/stats/pedigree.py

```diff
@@ -203,6 +203,23 @@ def _raise_on_half_founder(
             raise ValueError("Pedigree contains half-founders")
 
 
+@numba_jit
+def _identify_founders(
+    parent: ArrayLike, tau: ArrayLike = None
```

Comment:
I wonder if doing the control flow like this confuses numba jitting, as it's going to compile two versions here (one for tau=None and one for tau 2D array) I'd imagine.

Since it's such a simple function, perhaps it would be as well to do two versions, taking the control flow out of the tight inner loop?

Reply:
Fair call, it could even be inlined completely but it seems likely that the logic will be reused at some point